File: notes/coding/java/code/node.java
import java.util.*;

// Definition for a Node
class Node {
  public int val;
  public Node left;
  public Node right;
  // public Node parent;

  public Node() {}

  public Node(int val) {
    this.val = val;
    this.left = null;
    this.right = null;
  }

  public Node(int val, Node left, Node right) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

File: notes/coding/cpp/code/node.cpp
#include <bits/stdc++.h>

// Definition for a Node
class Node {
public:
  int val;
  Node *left;
  Node *right;
  // Node *parent;

  Node() {}

  Node(int val) {
    this->val = val;
    this->left = nullptr;
    this->right = nullptr;
  }

  Node(int val, Node *left, Node *right) {
    this->val = val;
    this->left = left;
    this->right = right;
  }
};
